src/extractor/MappingExtractor.ts

```diff
@@ -10,68 +14,75 @@ export class MappingExtractor {
     elements: ProcessableElementDefinition[],
     fhir: fhirdefs.FHIRDefinitions
   ): ExportableMapping[] {
-    const mappings =
-      input.mapping?.map(m => {
-        const mapping = new ExportableMapping(m.identity);
-        mapping.source = input.name;
-        if (m.name) mapping.title = m.name;
-        if (m.uri) mapping.target = m.uri;
-        if (m.comment) mapping.description = m.comment;
-        return mapping;
-      }) ?? [];
-    elements.forEach(element => {
-      this.extractRules(element, mappings);
-    });
+    const mappings = this.extractMappings(input, elements);
 
     // Filter out mappings on the parent - only include mappings new to the profile
     // or mappings on the base definition with additional mapping rules
+    // TODO: Look for parents from the local package being processed. This requires a MasterFisher.
     const parent = fhir.fishForFHIR(
       input.baseDefinition,
       utils.Type.Resource,
       utils.Type.Type,
       utils.Type.Profile,
       utils.Type.Extension
     );
-    const newItems = differenceWith(input.mapping, parent?.mapping, isEqual);
-    const newMappings = mappings.filter(mapping => newItems.some(i => i.identity === mapping.name));
-    const inheritedMappings = mappings.filter(mapping =>
-      parent?.mapping.some((i: fhirtypes.StructureDefinitionMapping) => i.identity === mapping.name)
-    );
 
-    // For each inherited mapping, look to see if there are any new rules
-    const changedInheritedMappings = inheritedMappings.filter(mapping => {
+    // If there is no parent found, all the mappings should be exported.
+    if (parent == null || !StructureDefinitionProcessor.isProcessableStructureDefinition(parent))
+      return mappings;
+
+    const parentSDElements =
+      parent.snapshot?.element?.map(rawElement => {
+        return ProcessableElementDefinition.fromJSON(rawElement, false);
+      }) ?? [];
+
+    const parentMappings = this.extractMappings(parent, parentSDElements);
+
+    // Filter out any mappings that come directly from the parent SD.
+    // Only return mappings that are new to the profile or inherited mappings that include new MappingRules
+    const newMappings = mappings.filter(mapping => {
```

Comment:
This feels a little weird to me because inside the filter, you are _mutating_ the input `mappings`.  This isn't something we usually expect from a filter operation...  It's valid, but... I don't know... it just feels a little surprising to me.

You _could_ refactor so it has:
```ts
const unmodifiedMappingIndexes: number[] = [];
```
then iterate `mappings.forEach` and push to the `unmodifiedMappingIndexes` any mappings that are 100% inherited from the parent.  Finally `pullAt` those indexes from `mappings`.  You are still mutating `mappings`, but since it is happening in `forEach` and since there is no additional `newMappings` array, it seems fine to mutate in this context.

Or ignore me if I'm just hung up on mutating in a filter function and I need to get over it.

Reply:
I tried to refactor this in a branch called [extract-mapping-filter-refactor](https://github.com/FHIR/GoFSH/tree/extract-mapping-filter-refactor). The last commit is the only one different from this branch. However, I'm a little confused by how you were suggesting to refactor this, so I'm not sure if that commit is getting at the spirit of what you were suggesting. If you already have your suggestion written out, feel free to commit it to that branch. Or I'm happy to chat on the phone if it is easier at this point.

Reply:
That wasn't exactly what I had in mind, but when I implemented what I had in mind, I didn't like it (bad idea "Yesterday Chris").  But instead, I came up with this -- yet another approach (on branch [extract-mapping-filter-refactor-cmm](https://github.com/FHIR/GoFSH/tree/extract-mapping-filter-refactor-cmm)): https://github.com/FHIR/GoFSH/commit/021330fdd01059d08138591c80fa53e3e6fba7c5

Reply:
BTW -- my commit above is not the most efficient (it loops through rules even when there is no parent) -- but I'm not too concerned about efficiency and it looked much prettier that way than my more efficient attempt. ;-)

Reply:
Thanks for looking at it again. I guess I'm still struggling to understand why this approach is better than the current approach since the original `mappings` array is modified in both cases. But I guess it's just the difference between mutating `mappings` within a `forEach` vs within a `filter`. Is that right? If so, I'm comfortable using your implementation on this branch.

Reply:
Yeah.  It's basically that you don't expect a _filter_ operation to have side-effects.  It should have one job and one job only -- to narrow a set down to a (potentially) smaller set.  A _forEach_ however, is much more general in nature and using a _forEach_ to mutate objects is a common use case -- so no one should be "surprised" by it.

Reply:
Okay! Then I shall push your implementation. Thanks for looking at it and coming up with a solution!

Reply:
I feel kind of like a GitHub bully pushing my solution on you.  I'm sorry!

Reply:
Here it is! 23a679e

And you are not a GitHub bully!! You have helped me learn something and you simplified something that I've stared at for too long and couldn't simplify!